discord/ext/commands/bot.py

```diff
@@ -795,26 +814,26 @@ def reload_extension(self, name, *, package=None):
             # if the load failed, the remnants should have been
             # cleaned from the load_extension function call
             # so let's load it from our old compiled library.
-            lib.setup(self)
+            lib.setup(self)  # type: ignore
```

Comment:
Unless we add a HasSetup protocol, we'll need to type-ignore `setup` or do something simular to what is done in `_call_module_finalizers`